src/modules/endpoint/hocs/withFetcher.js

```diff
@@ -1,127 +1,40 @@
 import style from './withFetcher.css'
-import { getContext } from 'recompose'
-import Endpoint from '../Endpoint'
-import { isEqual } from 'lodash/lang'
-
-const contextTypes = {
-  getEndpoint: PropTypes.func
-}
-
-/**
- * This function builds a HOC from a component
- * @callback buildHOC
- * @param {React.Component} WrappedComponent
- * @return {React.Component} The HOC
- */
-
-/**
- * This creates a factory for a Higher-Order-Component. The HOC attaches a fetcher to the supplied component.
- * @param endpointName {string} The name of the endpoint to fetch from
- * @param FailureComponent {*} the component which is rendered in error-case, null if no Failure should be rendered.
- * @param hideSpinner {boolean} If you want to hide a loading spinner in the render() method
- * @return {buildHOC} Returns a HOC which renders the supplied component as soon as the fetcher succeeded
- */
-export function withFetcher (endpointName, FailureComponent = Failure, hideSpinner = false) {
-  return WrappedComponent => {
-    class Fetcher extends React.Component {
-      static displayName = `${endpointName}Fetcher`
-      static propTypes = {
-        state: PropTypes.object.isRequired,
-        endpoint: PropTypes.instanceOf(Endpoint).isRequired,
-        requestAction: PropTypes.func.isRequired
-      }
-
-      static contextTypes = contextTypes
-
-      constructor () {
-        super()
-        this.state = {isDataAvailable: false}
-      }
-
-      componentWillMount () {
-        // A store dispatch in the componentWillMount (and therefore a state update) does not cause the props to update
-        // before render() in the Fetcher component is called the first time.
-        // This causes the <WrappedComponent> to be displayed with outdated props.
-        // https://github.com/reactjs/react-redux/issues/210#issuecomment-166055644
-        this.tryRequest()
-      }
-
-      componentWillReceiveProps (nextProps) {
-        // Dispatch new requestAction to ask the endpoint whether data is available, if:
-        // (a) the Fetcher endpoint.payloadName prop changed because of new data in the store (e.g. because a payload has been fetched)
-        if (!isEqual(this.props.state, nextProps.state) || this.getStatePayload(this.props) !== this.getStatePayload(nextProps)) {
-          this.tryRequest()
-        }
-      }
 
-      getStatePayload (props = this.props) {
-        return props.state[this.props.endpoint.stateName]
-      }
-
-      /**
-       * Tries to trigger a request by dispatching a requestAction
-       */
-      tryRequest () {
-        const storeResponse = this.props.requestAction()
-        this.setState({isDataAvailable: storeResponse.dataAvailable})
-      }
-
-      render () {
-        const payload = this.getStatePayload()
+const withFetcher = (name, urlMapper, mapper, params, FailureComponent = Failure) => WrappedComponent =>
+  class Fetcher extends React.Component {
+    constructor (props) {
+      super(props)
```

Comment:
You don't need to pass the props afaik. You can just remove them from the super and constructor.